diff --git a/lib/index.d.ts b/lib/index.d.ts
index 4a57f68ebb657e0c96352a959fc0ea225e802af5..01931f3b4c31f56fd40923e9aea5bd814b9f9864 100644
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -4,14 +4,14 @@
  *
  * @param {Nodes} node
  *   Tree to search.
- * @param {string} name
+ * @param {string|RegExp} name
  *   Comment name to look for.
  * @param {Handler} handler
  *   Handle a section.
  * @returns {undefined}
  *   Nothing.
  */
-export function zone(node: Nodes, name: string, handler: Handler): undefined;
+export function zone(node: Nodes, name: string | RegExp, handler: Handler): undefined;
 export type Nodes = import('mdast').Nodes;
 export type Parents = import('mdast').Parents;
 /**
diff --git a/lib/index.js b/lib/index.js
index d9d977bdac9b30c3aba2b5c6ecc12a0b79a18568..a967b0cb24db91f0f6ef803c987fd17579373b9d 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -40,7 +40,7 @@ import {visit} from 'unist-util-visit'
  *
  * @param {Nodes} node
  *   Tree to search.
- * @param {string} name
+ * @param {string|RegExp} name
  *   Comment name to look for.
  * @param {Handler} handler
  *   Handle a section.
@@ -58,7 +58,8 @@ export function zone(node, name, handler) {
   visit(node, function (node, index, parent) {
     const info = commentMarker(node)
     const match =
-      info && info.name === name
+      info &&
+      (typeof name === 'string' ? info.name === name : info.name.match(name))
         ? info.attributes.match(/(start|end)\b/)
         : undefined
     const type = match ? match[0] : undefined
